File: src/main/java/com/sumup/purecode/users/UserService.java
package com.sumup.purecode.users;

import jakarta.persistence.EntityNotFoundException;
import org.springframework.stereotype.Service;

@Service
public class UserService {

    private final UserRepository userRepository;

    public UserService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public UserDto getById(Long id) {
        return userRepository
                .findById(id)
                .map(UserDto::fromEntity)
                .orElseThrow(() -> new EntityNotFoundException("User " + id + " not found"));
    }
}

File: src/main/java/com/sumup/purecode/users/UserService.kt
package com.sumup.purecode.users

import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service

@Service
class UserService(private val userRepository: UserRepository) {

    fun getById(id: Long): UserDto? {
        return userRepository.findByIdOrNull(id)
            ?.let(UserDto::fromEntity)
    }
}
